lib/presentation/home/widgets/password_modal.dart

```diff
@@ -0,0 +1,188 @@
+                enabledBorder: OutlineInputBorder(
+                  borderRadius: BorderRadius.circular(20.0),
+                  borderSide:
+                      const BorderSide(width: 0, color: Colors.transparent),
+                ),
+                focusedBorder: OutlineInputBorder(
+                  borderRadius: BorderRadius.circular(20.0),
+                  borderSide:
+                      const BorderSide(width: 0, color: Colors.transparent),
+                ),
+                focusedErrorBorder: OutlineInputBorder(
+                  borderRadius: BorderRadius.circular(20.0),
+                  borderSide: const BorderSide(width: 2, color: Colors.red),
+                ),
+                errorText: _validated != null && _validated == false
+                    ? "Invalid password"
+                    : null,
+                errorBorder: OutlineInputBorder(
+                  borderRadius: BorderRadius.circular(20.0),
+                  borderSide: const BorderSide(width: 2, color: Colors.red),
+                ),
+                suffixIcon: IconButton(
+                  icon: _showInput
+                      ? const Icon(CollactionIcons.eye)
+                      : const Icon(CollactionIcons.eyeOff),
+                  onPressed: () {
+                    setState(() {
+                      _showInput = !_showInput;
+                    });
+                  },
+                  splashRadius: 2,
+                ),
+              ),
+              obscuringCharacter: '*',
+              cursorColor: kAccentColor,
+            ),
+          ),
+          CircleAvatar(
+            backgroundColor: _disableButton ? kPrimaryColor200 : kAccentColor,
+            minRadius: 30,
+            child: IconButton(
+              onPressed: () {
+                if (_disableButton == true) {
+                  return;
+                } else {
+                  _validatePassword();
+                }
+              },
+              icon: const Icon(CollactionIcons.arrowRight),
+              color: kSecondaryColor,
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  void _validateInput(String value) {
+    setState(() {
+      value == "" ? _disableButton = true : _disableButton = false;
+    });
+  }
+
+  void _validatePassword() {
+    if (_controller.text == widget.crowdAction.passwordJoin) {
+      setState(() {
+        _validated = true;
+      });
+      addCrowdActionAccess();
+      context.router.pop();
+      context.router.push(
+        CrowdActionDetailsRoute(
+          crowdAction: widget.crowdAction,
+        ),
+      );
+    } else {
+      setState(() {
+        _validated = false;
+      });
+    }
+  }
+
+  Future<void> addCrowdActionAccess() async {
+    final _settingsRepository = getIt<ISettingsRepository>();
+    await _settingsRepository.addCrowdActionAccess(
+        crowdActionId: widget.crowdAction.crowdactionID);
+  }
```

Comment:
All of this logic, together with all `setState` that we have, should be encapuslated in a BLOC, so we don't have application logic in presentation layer.